cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,105 @@
 /// <reference types='cypress' />
 
 describe('Student Registration page', () => {
-  before(() => {
+  const { generateUser } = require('../support/generateUser');
 
+  beforeEach(() => {
+    cy.visit('/');
   });
 
-  it('', () => {
+  it('should fill the form and assert modal content', () => {
+    const randomData = generateUser();
+    // first name, last name
+    cy.get('[placeholder="First Name"]')
+      .type(randomData.firstName);
+    cy.get('[placeholder="Last Name"]')
+      .type(randomData.lastName);
+    // email
+    cy.get('[placeholder="name@example.com"]')
+      .type(randomData.email);
+    // gender radio
+    cy.get('label[for="gender-radio-1"]')
+      .click();
+    // mobile
+    cy.get('[placeholder="Mobile Number"]')
+      .type(randomData.phone);
+    // date of birth
+    cy.get('#dateOfBirthInput')
+      .click();
+    cy.get('.react-datepicker__month-select')
+      .select('5');
+    cy.get('.react-datepicker__year-select')
+      .select('2002');
+    cy.get('.react-datepicker__day--017')
+      .click();
+    // subjects
+    cy.get('#subjectsInput')
+      .type('Physics{enter}');
+    // hobbies checkbox
+    cy.get('label[for="hobbies-checkbox-1"]')
+      .click();
+    cy.get('label[for="hobbies-checkbox-3"]')
+      .click();
+    // address textfield
+    cy.get('[id="currentAddress"]')
```

Comment:
This is not a proper way of selecting element via the id. Use `#`.

Reply:
Fixed